Source/API/CBLReplication.m

```diff
@@ -343,7 +343,7 @@ - (void) updateStatus: (CBLReplicationStatus)status
         return;
     if (status == kCBLReplicationStopped) {
         _started = NO;
-        [_database forgetReplication: self];
+        [self forgetReplication];
```

Comment:
Oh, I see. Yes, moving that block up above the async call sounds like a good fix.